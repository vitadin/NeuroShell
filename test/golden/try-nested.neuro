%% Test nested \try commands to verify proper error handling
\set[_echo_command="true"]
%% This test covers \try \try scenarios and complex error propagation

%% Test successful nested try
\try \try \bash echo "nested success"
\get[@status]
\get[@error]
\get[_output]

%% Test nested try with inner failure
\try \try \bash ls /nonexistent_nested
\get[@status]
\get[@error]

%% Test nested try with unknown command
\try \try \unknown_nested_command
\get[@status]
\get[@error]

%% Test triple nesting
\try \try \try \set[nested_var=deep]
\get[@status]
\get[nested_var]

%% Test nested try with outer failure (try of unknown command)
\try \unknown_outer_command \try \bash echo "never executed"
\get[@status]
\get[@error]