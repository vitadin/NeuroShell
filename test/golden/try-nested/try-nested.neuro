%% Test nested \try commands to verify proper error handling
\set[_echo_commands="true"]
%% This test covers \try \try scenarios and complex error propagation

%% Test successful nested try
\try \try \bash echo "nested success"
\get[_status]
\get[_error]
\get[_output]

%% Test nested try with inner failure
\try \try \bash ls /nonexistent_nested
\get[_status]
\get[_error]

%% Test nested try with unknown command
\try \try \unknown_nested_command
\get[_status]
\get[_error]

%% Test triple nesting
\try \try \try \set[nested_var=deep]
\get[_status]
\get[nested_var]

%% Test nested try with outer failure (try of unknown command)
\try \unknown_outer_command \try \bash echo "never executed"
\get[_status]
\get[_error]