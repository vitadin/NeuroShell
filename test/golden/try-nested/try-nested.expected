Setting _echo_command = true
%%> \bash echo "nested success"
nested success
%%> \get[_status]
_status = 0
%%> \get[_error]
_error = 
%%> \get[_output]
_output = nested success
%%> \bash ls /nonexistent_nested
Error: ls: /nonexistent_nested: No such file or directory
Exit status: 1
%%> \get[_status]
_status = 1
%%> \get[_error]
_error = ls: /nonexistent_nested: No such file or directory
%%> \get[_status]
_status = 1
%%> \get[_error]
_error = unknown command: unknown_nested_command
%%> \set[nested_var=deep]
Setting nested_var = deep
%%> \get[_status]
_status = 0
%%> \get[nested_var]
nested_var = deep
%%> \get[_status]
_status = 1
%%> \get[_error]
_error = unknown command: unknown_outer_command